tools/LoadPackage.groovy

```diff
@@ -0,0 +1,283 @@
+import org.ow2.proactive.authentication.ConnectionInfo
+import org.ow2.proactive.scheduler.rest.SchedulerClient
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+import org.apache.log4j.Logger
+
+import java.util.zip.ZipFile
+
+
+class LoadPackage {
+
+    private final String PATH_TO_SCHEDULER_CREDENTIALS_FILE = "config/authentication/scheduler.cred"
+
+    private final String LOAD_PACKAGE_SCRIPT_NAME = "LoadPackage.groovy"
+
+    private final String BUCKET_OWNER = "GROUP:public-objects"
+
+    private final String GLOBAL_SPACE_PATH
+    private final String SCHEDULER_REST_URL
+    private final String SCHEDULER_HOME
+    private final File WORKFLOW_TEMPLATES_DIR
+    private final String WORKFLOW_TEMPLATES_DIR_PATH
+    private final String CATALOG_URL
+
+
+
+    private logger = Logger.getLogger("org.ow2.proactive.scheduler")
+
+
+    LoadPackage(binding) {
+
+        // Bindings
+        this.GLOBAL_SPACE_PATH = binding.variables.get("pa.scheduler.dataspace.defaultglobal.localpath")
+        this.SCHEDULER_REST_URL = binding.variables.get("pa.scheduler.rest.url")
+        this.SCHEDULER_HOME = binding.variables.get("pa.scheduler.home")
+
+        // User variables
+        this.WORKFLOW_TEMPLATES_DIR = new File(this.SCHEDULER_HOME, "config/workflows/templates")
```

Comment:
Currently this script makes the assumption that it's being executed on the server, and with the same account. The client is supposed to be allowed running on any machine.

If the need of this change is to really allow a package to be installed from a remote client, then some logic must be changed. For example, a rest endpoint could be created to allow the execution of a goorvy script on the server

Reply:
You're are right Fabien. I know this limitation, which is ok at the moment (hub depends on it).
Then, next step is to offer this capacity from the REST API.